frontend/src/components/GoogleMaps/GooglePlacesInput.tsx

```diff
@@ -1,43 +1,52 @@
 import Geocoder from "react-native-geocoding";
 import React, { useState, useEffect, useRef } from "react";
-import MapView, { PROVIDER_GOOGLE, Circle } from "react-native-maps";
+import MapView, { Circle } from "react-native-maps";
 import { StyleSheet, View, Dimensions, Text } from "react-native";
 import Slider from "@react-native-community/slider";
 import { GooglePlacesInputProps } from "@/types";
 
 // env
-const GOOGLE_MAP_API_KEY = process.env.EXPO_PUBLIC_GOOGLE_MAP_API_KEY || "undefined";
+const GOOGLE_MAP_API_KEY =
+  process.env.EXPO_PUBLIC_GOOGLE_MAP_API_KEY || "undefined";
 import { LocationSearch } from "./LocationSearch";
-import Geocoding from 'react-native-geocoding';
 
-const { width, height } = Dimensions.get("window");
-Geocoding.init(GOOGLE_MAP_API_KEY);
+const { width } = Dimensions.get("window");
 
-const GooglePlacesInput = ({ onLocationSelect, value }: GooglePlacesInputProps) => {
+const GooglePlacesInput = ({
+  onLocationSelect,
+  value,
+}: GooglePlacesInputProps) => {
   Geocoder.init(GOOGLE_MAP_API_KEY); // use a valid API key
   const [query, setQuery] = useState({ address: "", citystate: "" });
   const [centerCircle, setCenterCircle] = useState("");
-  const [coord, setCoord] = useState<{ latitude: number; longitude: number; }>({
+  const [coord, setCoord] = useState<{ latitude: number; longitude: number }>({
     latitude: 37.733795,
     longitude: -122.446747,
   });
   const mapRef = useRef(null);
   const [radius, setRadius] = useState(30000);
+  const [sliderTimeoutId, setSliderTimeoutId] = useState<NodeJS.Timeout | null>(
+    null,
+  );
   const handleSliderChange = (value: number) => {
-    setRadius(value);
+    if (sliderTimeoutId) clearTimeout(sliderTimeoutId);
```

Comment:
This is good example of a debounce. I think the solution can be more simple though. Take a look at `onSlidingComplete` from their [doc](https://www.npmjs.com/package/@react-native-community/slider)